[Elastic Agent] Update diagnostics structure and document how to disable output status reporting (#4155)
elastic/docs-content: reference/fleet/agent-command-reference.md

Gather diagnostics information from the {{agent}} and component/unit it’s running. This command produces an archive that contains:

* version.txt - version information

> * agent-info.yaml - Contains the agent local metadata that is also sent to Fleet.

Reviewer: for consistency with the rest of the list items
```suggestion
* agent-info.yaml - contains the agent local metadata that is also sent to Fleet
```

* pre-config.yaml - pre-configuration before variable substitution - this is the elastic-agent.yaml from disk or from Fleet
* otel.yaml - the OpenTelemetry collector configuration file used with the `otel` sub-command
* otel-merged.yaml - the final configuration file provided to the OpenTelemetry collector including any internally generated collector components.
* variables.yaml - current variable contexts from providers
* environment.yaml - current environment variables visible to the Elastic Agent process.
* computed-config.yaml - configuration after variable substitution
* components-expected.yaml - expected computed components model from the computed-config.yaml
* components-actual.yaml - actual running components model as reported by the runtime manager
* state.yaml - current state information of all running components

> * *.pprof.gz files from the running elastic-agent process for analysis with `go tool pprof`:

Reviewer: ```suggestion
* *.pprof.gz files from the running `elastic-agent` process for analysis with `go tool pprof`:
```

    * goroutine.pprof.gz - goroutine dump
    * heap.pprof.gz - memory allocation of live objects
    * allocs.pprof.gz - sampling past memory allocations
    * threadcreate.pprof.gz - traces led to creation of new OS threads
    * block.pprof.gz - stack traces that led to blocking on synchronization primitives
    * mutex.pprof.gz - stack traces of holders of contended mutexes
* components/ directory - diagnostic information from each running component process:
    * Typically one directory per input-output pair representing a supervised process containing:
      * *.pprof.gz - profiles for analysis with `go tool pprof`; refer to the previous descriptions
      * *_metrics.json - metrics snapshots captured from running Beat processes

> * edot/ directory - the output of the elasticsdiagnostics collector extension:

Author: ```suggestion
* edot/ directory - the output of Elastic's collector diagnostics extension:
```

>     * otel-merged-actual.yaml - the configuration the collector is currently running with which should match otel-merged.yaml above

Reviewer: ```suggestion
    * otel-merged-actual.yaml - the configuration the collector is currently running with; it should match the otel-merged.yaml file described previously
```
